src/plugins/KartesianProfile.cpp

```diff
@@ -195,21 +258,69 @@ void KartesianProfile::endStep(ParticleContainer *particleContainer, DomainDecom
+    auto uID = (unsigned long) (xun * samplInfo.universalProfileUnit[1] * samplInfo.universalProfileUnit[2]
+                           + yun * samplInfo.universalProfileUnit[2] + zun);
     return uID;
 }
 
+long KartesianProfile::getCylUID(ParticleIterator &thismol) {
+
+    int phiUn,rUn,hUn;// (phiUn,rUn,yun): bin number in a special direction, e.g. rUn==5 corresponds to the 5th bin in the radial direction,
+    long unID;	// as usual
+    double xc,yc,zc; // distance of a particle with respect to the origin of the cylindrical coordinate system
+
+    unID = -1; // initialization, causes an error message, if unID is not calculated in this method but used in record profile
+
+    xc = thismol->r(0) - samplInfo.universalCentre[0];
+    yc = thismol->r(1) - samplInfo.universalCentre[1];
+    zc = thismol->r(2) - samplInfo.universalCentre[2];
+
+    // transformation in polar coordinates
+    double R2 = xc*xc + zc*zc;
+    // TODO: CHANGED TO R
+    //double R = sqrt(R2);
+    double phi = atan2(zc, xc); // asin(zc/sqrt(R2)) + ((xc>=0.0) ? 0:M_PI);
+    if(phi<0.0) {phi = phi + 2.0*M_PI;}
+
+    rUn = (int)floor(R2 * samplInfo.universalInvProfileUnit[0]);   // bin no. in R-direction
+    hUn = (int)floor(yc *  samplInfo.universalInvProfileUnit[1]);   // bin no. in H-direction
+    phiUn = (int)floor(phi *  samplInfo.universalInvProfileUnit[2]);   // bin no. in Phi-direction
+
+    // Check if R is inside cylinder
+    if(rUn >= (int)samplInfo.universalProfileUnit[0])   {
+        return -1;
+    }
+
+    if((rUn >= 0) && (hUn >= 0) && (phiUn >= 0) &&
+       (rUn < (int)samplInfo.universalProfileUnit[0]) && (hUn < (int)samplInfo.universalProfileUnit[1]) && (phiUn < (int)samplInfo.universalProfileUnit[2]))
+    {
+        unID = (long) (hUn * samplInfo.universalProfileUnit[0] * samplInfo.universalProfileUnit[2]
+               + rUn * samplInfo.universalProfileUnit[2] + phiUn);
+    }
+    else
+    {
+        global_log->error() << "INV PROFILE UNITS " << samplInfo.universalInvProfileUnit[0] << " " << samplInfo.universalInvProfileUnit[1] << " " << samplInfo.universalInvProfileUnit[2] << "\n";
+        global_log->error() << "PROFILE UNITS " << samplInfo.universalProfileUnit[0] << " " << samplInfo.universalProfileUnit[1] << " " << samplInfo.universalProfileUnit[2] << "\n";
+        global_log->error() << "Severe error!! Invalid profile ID (" << rUn << " / " << hUn << " / " << phiUn << ").\n\n";
+        global_log->error() << "Severe error!! Invalid profile unit (" << R2 << " / " << yc << " / " << phi << ").\n\n";
+        global_log->error() << "Coordinates off center (" << xc << " / " << yc << " / " << zc << ").\n";
+        global_log->error() << "unID = " << unID << "\n";
+        Simulation::exit(707);
+    }
+    return unID;
+}
+
 void KartesianProfile::addProfile(ProfileBase *profile) {
-    profile->init(this);
+    global_log->info() << "PROFILE ADD\n";

```

Comment:
change "PROFILE ADD" to "Profile added:" or something; i.e. fix the caps

Reply:
changed, e5ff251